src/software/ai/hl/stp/tactic/dribble/dribble_fsm_test.cpp

```diff
@@ -0,0 +1,66 @@
+#include "software/ai/hl/stp/tactic/dribble/dribble_fsm.h"
+
+#include <gtest/gtest.h>
+
+#include "software/test_util/test_util.h"
+
+TEST(DribbleFSMTest, test_transitions)
+{
+    Robot robot = ::TestUtil::createRobotAtPos(Point(-2, -3));
+    World world = ::TestUtil::createBlankTestingWorld();
+    world =
+        ::TestUtil::setBallPosition(world, Point(0.5, 0), Timestamp::fromSeconds(123));
+    world =
+        ::TestUtil::setBallVelocity(world, Vector(0, -1), Timestamp::fromSeconds(123));
+
+    HFSM<DribbleFSM> fsm;
+
+    // Start in DribbleState
+    EXPECT_TRUE(fsm.is(boost::sml::state<DribbleFSM::GetPossessionState>));
+
+    // Stay in DribbleState since ball not in possession yet
+    fsm.process_event(
+        DribbleFSM::Update({std::nullopt, std::nullopt, false},
+                           TacticUpdate(robot, world, [](std::unique_ptr<Intent>) {})));
+    EXPECT_TRUE(fsm.is(boost::sml::state<DribbleFSM::GetPossessionState>));
+
+    // At ball point so transition to done
```

Comment:
Should this comment say "transition to dribble state"?

Reply:
fixed